### PR TITLE
Change password functionality

### PR DESCRIPTION
**Pull request by crufter**
- [x] Change secret endpoint
- [x] Change password CLI command
- [x] Test

### CODE REVIEW

File: client/cli/user/user.go [hunk 1]
```diff
@@ -37,11 +44,96 @@ func init() {
...
+			if pw != pwVer {
+				fmt.Println("Passwords do not match. Please try again.")
+				continue
+			}
+			newPassword = pw
+			break
+		}
+	}
+
+	// Inspect the token
```
Comment (ben-toogood) [inline client/cli/user/user.go:117]: Why do we not do this before asking for passwords?

File: client/cli/user/user.go [hunk 2]
```diff
@@ -37,11 +44,96 @@ func init() {
...
+	}
+
+	accountService := pb.NewAccountsService("auth", client.DefaultClient)
+	_, err = accountService.ChangePassword(context.TODO(), &pb.ChangePasswordRequest{
+		Id:        acc.ID,
+		OldSecret: oldPassword,
+		NewSecret: newPassword,
+	}, goclient.WithAuthToken())
+	if err != nil {
+		fmt.Println(err)
```
Comment (ben-toogood) [inline client/cli/user/user.go:131]: Rather than printing the error and exiting, you can just return the error.

File: client/cli/user/user.go [hunk 3]
```diff
@@ -37,11 +44,96 @@ func init() {
...
+	// Inspect the token
+	acc, err := auth.Inspect(token)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	accountService := pb.NewAccountsService("auth", client.DefaultClient)
+	_, err = accountService.ChangePassword(context.TODO(), &pb.ChangePasswordRequest{
+		Id:        acc.ID,
```
Comment (ben-toogood) [inline client/cli/user/user.go:126]: This should probably be determined using the account.FromContext() function in the handler, as apposed to be passed from the client

File: service/auth/proto/auth.proto [hunk 4]
```diff
@@ -147,4 +148,14 @@ message ListRequest {
 
 message ListResponse {
 	repeated Rule rules = 1;
+	string a = 2;
```
Comment (ben-toogood) [inline service/auth/proto/auth.proto:151]: I think this is a typo?

File: client/cli/user/user.go (hunk 1 above)
Comment (crufter) [reply client/cli/user/user.go:117]: Good point

File: service/auth/proto/auth.proto [hunk 5]
```diff
@@ -147,4 +148,14 @@ message ListRequest {
...
 }
+
+message ChangePasswordRequest{
```
Comment (ben-toogood) [inline service/auth/proto/auth.proto:154]: Mixed terminology: password vs secret.

File: client/cli/user/user.go (hunk 3 above)
Comment (crufter) [reply client/cli/user/user.go:126]: I will look that up thanks.

File: service/auth/server/auth/accounts.go [hunk 6]
```diff
@@ -115,6 +116,55 @@ func (a *Auth) Delete(ctx context.Context, req *pb.DeleteAccountRequest, rsp *pb
...
+// ChangePassword by providing a refresh token and a new secret
+func (a *Auth) ChangePassword(ctx context.Context, req *pb.ChangePasswordRequest, rsp *pb.ChangePasswordResponse) error {
+	// set defaults
+	if req.Options == nil {
+		req.Options = &pb.Options{}
+	}
+	if len(req.Options.Namespace) == 0 {
+		req.Options.Namespace = namespace.DefaultNamespace
+	}
+
```
Comment (ben-toogood) [inline service/auth/server/auth/accounts.go:128]: You need to authorise access to the namespace using the namespace.Authorize function. See List for example usage.

File: service/auth/proto/auth.proto (hunk 4 above)
Comment (crufter) [reply service/auth/proto/auth.proto:151]: yeah when i was trying to get proto working... fixing.

File: service/auth/proto/auth.proto (hunk 5 above)
Comment (crufter) [reply service/auth/proto/auth.proto:154]: Yeah makes sense.

File: service/auth/server/auth/accounts.go (hunk 6 above)
Comment (crufter) [reply service/auth/server/auth/accounts.go:128]: +1

File: client/cli/user/user.go (hunk 1 above)
Comment (crufter) [reply client/cli/user/user.go]: Changed based on our discussion

File: service/auth/proto/auth.proto (hunk 5 above)
Comment (crufter) [reply service/auth/proto/auth.proto]: Changed

File: service/auth/server/auth/accounts.go (hunk 6 above)
Comment (crufter) [reply service/auth/server/auth/accounts.go:128]: Done

**Pull request merged**